lib/nostrum/struct/overwrite.ex

```diff
@@ -14,7 +16,7 @@ defmodule Nostrum.Struct.Overwrite do
   ]
 
   @typedoc "Role or User id"
-  @type id :: Snowflake.t()
+  @type id :: Role.id() | User.id()
```

Comment:
I think this is better left the way we had it. For things like `guild_id` in the same module and throughout other modules, we don't link to `Guild.id()`, but to `Snowflake.t()` instead.

I think for consistency we should leave it how it was or change all other types lol